miner/worker.go

```diff
@@ -117,22 +117,30 @@ func (w *worker) commitNewWork() (*types.Block, error) {
 	defer w.mu.RUnlock()
 
 	tstart := w.clock.Time()
-	timestamp := tstart.Unix()
+	timestamp := uint64(tstart.Unix())
 	parent := w.chain.CurrentBlock()
 	// Note: in order to support asynchronous block production, blocks are allowed to have
 	// the same timestamp as their parent. This allows more than one block to be produced
 	// per second.
-	if parent.Time() >= uint64(timestamp) {
-		timestamp = int64(parent.Time())
+	if parent.Time() >= timestamp {
+		timestamp = parent.Time()
 	}
 
+	bigTimestamp := new(big.Int).SetUint64(timestamp)
 	var gasLimit uint64
+	// We need to get the fee config from the parent block because the fee config is
+	// not yet set in the current block.
+	// This is because the fee config might be set with the state in the FeeManager precompile.
+	// Which could change the fee configs in the current block.
+	// But we don't we don't know what the state will be for the current block.
+	// meaning that the tx that changes the precompile state itself could be also invalidated by the
+	// fee config that it sets.
```

Comment:
```suggestion
	// The fee config manager relies on the state of the parent block to set the fee config
	// because the fee config may be changed by the current block.
```